classic-sdk/core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/AvailabilityData.java

```diff
@@ -4,12 +4,13 @@
 package com.microsoft.applicationinsights.internal.schemav2;
 
 import com.microsoft.applicationinsights.telemetry.Duration;
+import javax.annotation.Nonnull;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
 public class AvailabilityData {
 
-  private String id;
+  @Nonnull private String id;
```

Comment:
I am tempted to set id by default
```
Private String id = UUID.randomUUID().toString()
```

Reply:
this class is not final and you can extend it and have your own id default to UUID. Does it work for you?

Reply:
Yes, it will work. However I was thinking if an agent can take care of creating a unique id, it will be easier for the end user.

Reply:
Some end users might not like to use UUID though. Will close the PR. Thanks.